### PR TITLE
Feature/select feedback

### PR DESCRIPTION
**Pull request by funrep**
Added visual feedback to selection in the form of resizing the crosshair.

### CODE REVIEW

**Comment by marcusrettig** [comment]
@funrep Nice! Do you also have time to also add a popup message? Maybe like [this](https://material.angular.io/components/snack-bar/examples) (press the button).

**Comment by funrep** [comment]
@marcusrettig good idea, I should do that!

**Comment by funrep** [comment]
@marcusrettig No, that's an error, closing this one and open a new.

**Comment by funrep** [comment]
@marcusrettig ping, added basic pop up, realized we need to make animations FPS-independent (integrate deltatime into the calculations), also there is refactoring to be made because Popup and Billboard share a bit of code, but I'll start a new branch for those changes.

**Comment by funrep** [comment]
@marcusrettig @perssonsimon1 could someone review this?

File: src/Selector.js [hunk 1]
```diff
@@ -6,6 +6,8 @@ export default class Selector {
   constructor() {
     this.onKeyPress = this.onKeyPress.bind(this);
     this.onMessage = this.onMessage.bind(this);
+    this.selectEvent = false;
```
Comment (marcusrettig) [inline src/Selector.js]: Could you name this differently? `selectEvent` doesn't make sense for a boolean.

File: src/Selector.js [hunk 2]
```diff
@@ -62,7 +74,11 @@ export default class Selector {
       else if (selected.isBillboard && selected.parent) {
         data = selected.parent.data;
       }
+      else if (selected.type === 'Mesh' && selected.parent.parent) {
+        data = selected.parent.parent.data;
```
Comment (marcusrettig) [inline src/Selector.js]: `data` will be undefined here. `selected.parent` will be a planet and `selected.parent.parent` the scene.

Comment (marcusrettig) [reply src/Selector.js]: Please also check in the if condition that the parent is actually a planet (use `.isPlanet`).

File: src/ui/Crosshair.js [hunk 3]
```diff
@@ -8,8 +8,44 @@ export default class Crosshair extends LineSegments {
     const geometry = createGeometry(0.05);
     const material = createMaterial();
 
-    super(geometry, material);
+    const obj = super(geometry, material);
```
Comment (marcusrettig) [inline src/ui/Crosshair.js]: Why is the result of calling `super` returned?

File: src/ui/UI.js [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
 
 export default class UI extends Object3D {
   constructor() {
     super();
 
     this.crosshair = new Crosshair();
-
     this.add(this.crosshair);
 
+    this.popup = new Popup('Added job!');
```
Comment (marcusrettig) [inline src/ui/UI.js]: `'Job saved!'` better? :)

File: src/ui/Crosshair.js [hunk 5]
```diff
@@ -8,8 +8,44 @@ export default class Crosshair extends LineSegments {
...
+    }
+    else if (this.selected) {
+      this.size = 0.05;
+      this.updateGeometry(this.size);
+      this.selected = false;
+      this.timer = 0;
+    }
   }
+
+  updateGeometry(size) {
```
Comment (marcusrettig) [inline src/ui/Crosshair.js]: Changing geometry should be avoided. Wouldn't scaling the crosshair have the save effect?

File: src/ui/Popup.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    this.selected = false;
+    this.isUp = false;
+    this.isMovingUp = true;
+    this.timer = 0;
+    this.timeLimit = 80;
+    this.add(this.sprite);
+  }
+
+  update(selector) {
+    // TODO: Rewrite this animation using tweens.
```
Comment (marcusrettig) [inline src/ui/Popup.js:26]: This works for now but in the future we should use tweens for these kinds of animations. I added a todo comment to remind us.

File: src/ui/Crosshair.js [hunk 7]
```diff
@@ -8,8 +8,44 @@ export default class Crosshair extends LineSegments {
...
+
+    this.size = 0.05;
+    this.selected = false;
+    this.timer = 0;
+    this.timeLimit = 30;
+    return obj;
+  }
+
+  update(selector) {
+    // TODO: Rewrite this animation using tweens.
```
Comment (marcusrettig) [inline src/ui/Crosshair.js]: This works for now but in the future we should use tweens for these kinds of animations. I added a todo comment to remind us.

**Review by marcusrettig — CHANGES_REQUESTED** [review]
See line comments. Most important to fix is in `Selector.js:78`.

File: src/Selector.js [hunk 8]
```diff
@@ -28,6 +30,14 @@ export default class Selector {
...
       this.selected = null;
     }
+
+    if (this.selectEvent && this.timer < 5) {
+      this.timer += 1;
+    }
+    else if (this.selectEvent && this.timer >= 5) {
+      this.timer = 0;
+      this.selectEvent = false;
+    }
```
Comment (marcusrettig) [inline src/Selector.js]: Not clear why the above if/else blocks are needed. Could you add a comment?

Comment (funrep) [reply src/Selector.js]: The `selectEvent` has to be true for the next frame for it to make effect, so I used the timer convention I used elsewhere here as well, it probably makes more sense having two boolean values instead.

**Review by marcusrettig — APPROVED** [review]

**Pull request merged**